### DISCUSSION TITLE
How to get the finger print of x509 Certificate

### DISCUSSION THREAD

**Discussion by divya-clari** [PeculiarVentures/x509#34] [General]
I am looking for a way to get the fingerprint of X509 Certificate. IS there an example that i can follow ? To give little bit more context we are migrating a legacy module from Rails. So Trying to achieve the same thing as below:
`fingerprint=OpenSSL::Digest::SHA1.new(Base64.decode64(x509Certificate)).to_s.scan(/../).map{ |s| s.upcase }.join(":")`

Expected FingerPrint pattern: "10A:12B:1A:C8...........:97"

**Comment by rmhrisk** [comment]
See the source for https://understandingwebpki.com/?cert=MIIFazCCA1OgAwIBAgIRAIIQz7DSQONZRGPgu2OCiwAwDQYJKoZIhvcNAQELBQAwTzELMAkGA1UEBhMCVVMxKTAnBgNVBAoTIEludGVybmV0IFNlY3VyaXR5IFJlc2VhcmNoIEdyb3VwMRUwEwYDVQQDEwxJU1JHIFJvb3QgWDEwHhcNMTUwNjA0MTEwNDM4WhcNMzUwNjA0MTEwNDM4WjBPMQswCQYDVQQGEwJVUzEpMCcGA1UEChMgSW50ZXJuZXQgU2VjdXJpdHkgUmVzZWFyY2ggR3JvdXAxFTATBgNVBAMTDElTUkcgUm9vdCBYMTCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIBAK3oJHP0FDfzm54rVygch77ct984kIxuPOZXoHj3dcKi%2FvVqbvYATyjb3miGbESTtrFj%2FRQSa78f0uoxmyF%2B0TM8ukj13Xnfs7j%2FEvEhmkvBioZxaUpmZmyPfjxwv60pIgbz5MDmgK7iS4%2B3mX6UA5%2FTR5d8mUgjU%2Bg4rk8Kb4Mu0UlXjIB0ttov0DiNewNwIRt18jA8%2Bo%2Bu3dpjq%2BsWT8KOEUt%2Bzwvo%2F7V3LvSye0rgTBIlDHCNAymg4VMk7BPZ7hm%2FELNKjD%2BJo2FR3qyHB5T0Y3HsLuJvW5iB4YlcNHlsdu87kGJ55tukmi8mxdAQ4Q7e2RCOFvu396j3x%2BUCB5iPNgiV5%2BI3lg02dZ77DnKxHZu8A%2FlJBdiB3QW0KtZB6awBdpUKD9jf1b0SHzUvKBds0pjBqAlkd25HN7rOrFleaJ1%2FctaJxQZBKT5ZPt0m9STJEadao0xAH0ahmbWnOlFuhjuefXKnEgV4We0%2BUXgVCwOPjdAvBbI%2Be0ocS3MFEvzG6uBQE3xDk3SzynTnjh8BCNAw1FtxNrQHusEwMFxIt4I7mKZ9YIqioymCzLq9gwQbooMDQaHWBfEbwrbwqHyGO0aoSCqI3Haadr8faqU9GY%2FrOPNk3sgrDQoo%2F%2Ffb4hVC1CLQJ13hef4Y53CIrU7m2Ys6xt0nUW7%2FvGT1M0NPAgMBAAGjQjBAMA4GA1UdDwEB%2FwQEAwIBBjAPBgNVHRMBAf8EBTADAQH%2FMB0GA1UdDgQWBBR5tFnme7bl5AFzgAiIyBpY9umbbjANBgkqhkiG9w0BAQsFAAOCAgEAVR9YqbyyqFDQDLHYGmkgJykIrGF1XIpu%2BILlaS%2FV9lZLubhzEFnTIZd%2B50xx%2B7LSYK05qAvqFyFWhfFQDlnrzuBZ6brJFe%2BGnY%2BEgPbk6ZGQ3BebYhtF8GaV0nxvwuo77x%2FPy9auJ%2FGpsMiu%2FX1%2BmvoiBOv%2F2X%2FqkSsisRcOj%2FKKNFtY2PwByVS5uCbMiogziUwthDyC3%2B6WVwW6LLv3xLfHTjuCvjHIInNzktHCgKQ5ORAzI4JMPJ%2BGslWYHb4phowim57iaztXOoJwTdwJx4nLCgdNbOhdjsnvzqvHu7UrTkXWStAmzOVyyghqpZXjFaH3pO3JLF%2Bl%2B%2F%2BsKAIuvtd7u%2BNxe5AW0wdeRlN8NwdCjNPElpzVmbUq4JUagEiuTDkHzsxHpFKVK7q4%2B63SM1N95R1NbdWhscdCb%2BZAJzVcoyi3B43njTOQ5yOf%2B1CceWxG1bQVs5ZufpsMljq4Ui0%2F1lvh%2BwjChP4kqKOJ2qxq4RgqsahDYVvTH9w7jXbyLeiNdd8XM2w9U%2Ft7y0Ff%2F9yi0GE44Za4rF2LN9d11TPAmRGunUHBcnWEvgJBQl9nJEiU0Zsnvgc%2FubhPgXRR4Xq37Z0j4r7g1SgEEzwxA57demyPxgcYxn%2FeR44%2FKJ4EBs%2BlVDR3veyJm%2BkXQ99b21%2F%2Bjh5Xos1AnX5iItreGCc%3D

**Comment by microshine** [comment]
You can compute the thumbprint using the `getThumbprint` method of the X509Certificate class.

https://peculiarventures.github.io/x509/classes/X509Certificate.html#getThumbprint

**NodeJS**
```js
const cert = new x509.X509Certificate(pem);
const thumbprint = await cert.getThumbprint();
const formattedThumbprint: string[] = [];
for (const octet of Buffer.from(thumbprint)) {
  formattedThumbprint.push(`${octet.toString(16)}`.padStart(2, "0").toUpperCase());
}
console.log(formattedThumbprint.join(":")); // 10:B9:24:B0:18:57:23:F2:2D:EB:FD:94:37:55:85:E9:CF:1D:8B:93
```

**Comment by Gereon99** [comment]
For some reason I couldn't use `Buffer`. I probably did something wrong, but I ended up with this solution that basically does the same thing. Maybe it helps someone in the future!

```javascript
const cert = new x509.X509Certificate(pem);
const thumbprint = await cert.getThumbprint();
const formattedThumbprint: string = Array.from(new Uint8Array(thumbprint), (byte) =>
    byte.toString(16).padStart(2, "0")
).join(":");
```
